File: MaximumPopulation.java
public class MaximumPopulation {
    public static void main(String[] args) {
        System.out.println(maximumPopulation(new int[][] {{1993, 1999}, {2000, 2010}}));
    }
    static int maximumPopulation(int[][] logs) {
        int[] freq = new int[2051];
        for (int[] log : logs) {
            int birth = log[0];
            int death = log[1];
            freq[birth]++;
            freq[death]--;
        }
        int max = 0;
        int year = 0;
        int current = 0;
        for (int i = 1950; i <= 2050; i++) {
            current += freq[i];
            if (current > max) {
                max = current;
                year = i;
            }
        }
        return year;
    }
}

File: MaximumPopulation.py

def maxPopulation(logs):
# Initialize a frequency array to store population changes
    freq = [0] * 2051
        
        # Iterate through each person's birth and death years
    for birth, death in logs:
        freq[birth] += 1  # Increment population for birth year
        freq[death] -= 1   # Decrement population for death year
        
        # Calculate prefix sum of the frequency array
    for i in range(1, len(freq)):
        freq[i] += freq[i - 1]
        
        # Find the year with the maximum population
    max_population = max(freq)
        
        # Find the earliest year(s) with maximum population
    earliest_year = freq.index(max_population)
        
    return earliest_year

print(maxPopulation(logs=[[1993, 1999], [2000, 2010]]))
print(maxPopulation(logs = [[1950,1961],[1960,1971],[1970,1981]]))